### ISSUE TITLE
long text files display white screen

### ISSUE THREAD

**Issue by pbstudent** [pressbooks/pressbooks#1458]
### Prerequisites

* [ X] Can you reproduce the behavior you wish to address with [all other plugins deactivated](http://codex.wordpress.org/Multisite_Network_Administration#Plugins)?
* [x ] Are you running the [latest version of Pressbooks](http://github.com/pressbooks/pressbooks/releases/latest/) and the [latest version of WordPress](http://codex.wordpress.org/Upgrading_WordPress)?

Pressbooks.com

* [ x] Are you experiencing this issue with one of our built-in root or book themes?
* [X ] Did you [perform a cursory search](https://github.com/issues?q=+is%3Aissue+repo%3Apressbooks%2Fpressbooks) to see if your bug or enhancement is already listed?

### Description
Opening a book page in the web editor that contains large quantity of text displays invisible text in editor.

### Steps to Reproduce

1. Add lots of text in editor
2. save page
3. go to Organize and reselect saved page
4. editor displays blank page
5. refresh web browser many times (up to 9x)
6. Text renders on page

**Expected behavior:** 

Return to editor view with displayed text

**Actual behavior:** 
editor section does not reform in size to the text and displays white text.

#### Book Info

Book ID: 102612
Book URL: http://collinscounselling.pressbooks.com/
Book Privacy: Public

#### Browser

Platform: OS X
Browser Name: Firefox
Browser Version: 63.0
User Agent String: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:63.0) Gecko/20100101 Firefox/63.0

#### WordPress Configuration

Network URL: http://pressbooks.com/
Network Type: Subdomain
Version: 4.9.8
Language: en_US
WP_ENV: production
WP_DEBUG: Disabled
Memory Limit: 64M

#### Pressbooks Configuration

Version: 5.5.5
Book Theme: McLuhan
Book Theme Version: 2.5.2
Root Theme: Pressbooks Main
Root Theme Version: 2.2.1

#### Pressbooks Dependencies

Epubcheck: Installed
Kindlegen: Installed
xmllint: Installed
PrinceXML: Not Installed
Saxon-HE: Installed

#### Must-Use Plugins

Bedrock Autoloader: 1.0.0
fix-recaptcha.php: n/a
hm-autoloader.php: n/a
redirect-ms-files.php: n/a
telemetry.php: n/a

#### Network Active Plugins

Akismet Anti-Spam: 4.0.8
Hypothesis: 0.5.0
Pressbooks: 5.5.5
Pressbooks BISAC: 2.0.0
Pressbooks Client Marketing: 1.0.1
Pressbooks Stats: 1.6.2
Pressbooks VIP: 5.3.0
Redis Object Cache: 1.3.8
User Activation Keys: 4.6
WP Fail2Ban Redux: 0.5.0

#### Book Active Plugins

Akismet Anti-Spam: 4.0.8
Hypothesis: 0.5.0
Pressbooks: 5.5.5
Pressbooks BISAC: 2.0.0
Pressbooks Client Marketing: 1.0.1
Pressbooks Stats: 1.6.2
Pressbooks VIP: 5.3.0
Redis Object Cache: 1.3.8
User Activation Keys: 4.6
WP Fail2Ban Redux: 0.5.0

#### Inactive Plugins

Lazy Load: 0.6.1
Pressbooks Whitelabel: 1.3.0
Redirection: 2.10.1
The SEO Framework: 2.9.4
WP QuickLaTeX: 3.8.5
WP reCaptcha Integration: 1.3.0

#### Server Configuration

PHP Version: 7.1.23-3+ubuntu16.04.1+deb.sury.org+1
MySQL Version: 5.6.10
Webserver Info: nginx/1.13.6

#### PHP Configuration

Safe Mode: Disabled
Memory Limit: 256M
Upload Max Size: 25M
Post Max Size: 25M
Upload Max Filesize: 25M
Time Limit: 180
Max Input Vars: 1000
URL-aware fopen: On (1)
Display Errors: N/A

#### PHP Extensions

OPcache: Zend
XDebug: Disabled
cURL: Supported
cURL Version: 7.47.0
imagick: Not Installed
xsl: Installed

[glossary.html.zip](https://github.com/pressbooks/pressbooks/files/2569235/glossary.html.zip)
![before_screen_refresh](https://user-images.githubusercontent.com/24277755/48311602-75494980-e5a2-11e8-9633-a52d05313aa6.png)
![after_screen_refresh](https://user-images.githubusercontent.com/24277755/48311604-78dcd080-e5a2-11e8-89d0-23a10e7b6ec8.png)

**Comment by greatislander** [comment]
@pbstudent My working theory is that the ongoing issue is a bug in WordPress core.

The WordPress-produced revisions table displayed below the post editor uses the same unoptimized query that I optimized in Pressbooks code. So, in order to display the list of revisions, WordPress is querying the database and loading the full content of all 158 revisions of the 43,000-word post in question. I've traced a couple of out-of-memory errors related to this operation on our server. I'm going to file a bug with WordPress and see if we can get this addressed upstream; in the meantime, your workaround seems like the best option.
